install-scripts/mac/skm-install-mac.sh

```diff
@@ -0,0 +1,128 @@
+	echo ""
+	if [[ $REPLY =~ [Yy]$ ]]
+	then
+		xcode-select --install
+	else
+    	echo "macOS developer tools not installed, please run: \"xcode-select --install\" in the terminal and then reinstall."
+		exit 1
+	fi
+}
+
+echo "Checking for macOS developer tools..."
+has_git
+if [ $? -ne 0 ]; then
+	install_git
+	has_git
+	if [ $? -ne 0 ]; then
+		printf "\t$MARK_CROSS Failed to install macOS developer tools. Please ensure you have \"git\" installed and try again.\n"
+		exit 1
+	fi
+else
+	printf "\t$MARK_CHECK macOS developer tools appears to be installed\n"
+fi
+
+echo "Checking for an existing version of SplashKit..."
+if [ -d "${INSTALL_PATH}" ]; then
+	printf "\t$MARK_CROSS SplashKit is already installed!\n"
+	read -p "Would you like to completely re-install SplashKit? " -n 1 -r
+	echo ""
+	if [[ $REPLY =~ [Yy]$ ]]; then
+		D=`date +%y%m%d-%H%M%S`
+		OLD_PATH="$INSTALL_PATH-$D"
+		echo "Removing existing SplashKit installation..."
+		mv $INSTALL_PATH $OLD_PATH
+		if [ -d "${INSTALL_PATH}" ]; then
+			printf "\t$MARK_CROSS failed to remove existing SplashKit installation.\n"
+			exit 1
+		else
+			printf "\t$MARK_CHECK removed existing SplashKit installation.\n"
+		fi
+	else
+		printf "\t$MARK_CHECK SplashKit is already installed.\n"
+		exit 1
+	fi
+fi
+
+echo "Installing SplashKit..."
+git clone --depth 1 --branch master $GIT_SKM_REPO "${INSTALL_PATH}"
+if [ -d "${INSTALL_PATH}" ]; then
+	printf "\t$MARK_CHECK SplashKit was installed.\n"
+else
+	printf "\t$MARK_CROSS failed to install SplashKit.\n"
+	exit 1
+fi
+
+# Add to .bashrc if using bash and path line is missing for SplashKit inclusion.
+if [ ${SHELL} = "/bin/bash" ]; then
+	echo "Adding \"$INSTALL_PATH\" to your bash profile..."
+	grep -Fqx "export PATH=\"$INSTALL_PATH:\$PATH\"" ~/.bash_profile || echo "export PATH=\"$INSTALL_PATH:\$PATH\"" >> ~/.bash_profile
+    grep -Fqx "export PATH=\"$INSTALL_PATH:\$PATH\"" ~/.bashrc || echo "export PATH=\"$INSTALL_PATH:\$PATH\"" >> ~/.bashrc
+fi
+
+# Add to .zshrc if using zsh and path line is missing for SplashKit inclusion.
+if [ ${SHELL} = "/bin/zsh" ]; then
+	echo "Adding \"$INSTALL_PATH\" to your ZSH profile..."
+    grep -Fqx "export PATH=\"$INSTALL_PATH:\$PATH\"" ~/.zshrc || echo "export PATH=\"$INSTALL_PATH:\$PATH\"" >> ~/.zshrc
+fi
+
+printf "\t$MARK_CHECK done\n"
+export PATH="$INSTALL_PATH:$PATH"
+
+# Verify installation and access
+SKM_PATH="${INSTALL_PATH}/skm"
+if [ -f $SKM_PATH ]; then
+	printf "$MARK_CHECK SplashKit installed successfully!\n"
+else
+	printf "$MARK_CROSS SplashKit installation failed!\n"
+fi
+
+find "${INSTALL_PATH}" -name "*.sh" -exec chmod a+x "{}" \;
+
+which skm 2>&1 > /dev/null
+if [ $? -ne 0 ]; then
+	printf "$MARK_CROSS Failed to validate \"skm\" command.\n"
+	exit 1
+else
+	printf "$MARK_CHECK Verified that \"skm\" can be accessed.\n"
+fi
+
+read -p "Would you like to install the recommended developer tools for SplashKit?: " -n 1 -r
+echo ""
+if [[ $REPLY =~ [Yy]$ ]]
```

Comment:
**suggestion:** Similar to the above suggestion ☝🏼, for unattended installation, this might be problematic as the script pauses waiting for user input. Additionally, I think the choice of IDE should be left with the developers. SKM, if possible, do not need to include this in the install script. For example, they can happily use Vim, Emacs to develop SplashKit app. What do you think?

Reply:
Fair enough re installing VS Code - but the website [0] does currently have it as a setup step. I will remove for now unless there are objections

[0] https://splashkit.io/articles/installation/mac/step-3/

Reply:
Also, on the unattended installation issue - I don't think the installer should start executing system commands such as `xcode-select` without prompting the user to confirm as this maybe trigger a license agreement request [0] the first time around. It may be confusing to users who aren't expecting it.

Regarding accepting user input, `bash <(curl script)` won't work, but `curl script | bash -e` or even `bash -c "$(curl script)" (like homebrew [1]) will.

[0] https://apple.stackexchange.com/questions/337744/installing-xcode-command-line-tools
[1] https://brew.sh/

Reply:
@isdampe Thank you for expanding on your thoughts 🙏🏼 

> but the website [0] does currently have it as a setup step. 

Interesting. I think we should just leave a link to the VSCode website on how to install the software. IMHO, it should be maintained by the VSCode team, and we should avoid having multiple sources of truth.

> Regarding accepting user input, bash <(curl script) won't work, but curl script | bash -e or even bash -c "$(curl script)" (like homebrew [1]) will.

So we might have to document this in the doc for the Mac installer if we would like to go down with this approach. Could you create a card to follow this up?